sklearn/ensemble/_hist_gradient_boosting/loss.py

```diff
@@ -123,34 +153,43 @@ class LeastSquares(BaseLoss):
 
         loss(x_i) = 0.5 * (y_true_i - raw_pred_i)**2
 
-    This actually computes the half least squares loss to optimize simplify
+    This actually computes the half least squares loss to simplify
     the computation of the gradients and get a unit hessian (and be consistent
     with what is done in LightGBM).
     """
+    def __init__(self, sample_weight):
+        # If sample weights are provided, the hessians and gradients
+        # are multiplied by sample_weight, which means the hessians are
+        # equal to sample weights.
+        super().__init__(hessians_are_constant=sample_weight is None)
 
-    hessians_are_constant = True
-
-    def __call__(self, y_true, raw_predictions, average=True):
+    def pointwise_loss(self, y_true, raw_predictions):
         # shape (1, n_samples) --> (n_samples,). reshape(-1) is more likely to
         # return a view.
         raw_predictions = raw_predictions.reshape(-1)
         loss = 0.5 * np.power(y_true - raw_predictions, 2)
-        return loss.mean() if average else loss
+        return loss
 
-    def get_baseline_prediction(self, y_train, prediction_dim):
-        return np.mean(y_train)
+    def get_baseline_prediction(self, y_train, sample_weight, prediction_dim):
+        return np.average(y_train, weights=sample_weight)
 
     @staticmethod
     def inverse_link_function(raw_predictions):
         return raw_predictions
 
     def update_gradients_and_hessians(self, gradients, hessians, y_true,
-                                      raw_predictions):
+                                      raw_predictions, sample_weight):
         # shape (1, n_samples) --> (n_samples,). reshape(-1) is more likely to
         # return a view.
         raw_predictions = raw_predictions.reshape(-1)
         gradients = gradients.reshape(-1)
-        _update_gradients_least_squares(gradients, y_true, raw_predictions)
+        hessians = hessians.reshape(-1)
```

Comment:
Move a few lines down to right above `_update_gradients_hessians_least_squares`. (Since it is only used by `_update_gradients_hessians_least_squares`)